### DISCUSSION TITLE
Fluent Grid conditional column content

### DISCUSSION THREAD

**Discussion by madcoda9000** [microsoft/fluentui-blazor#704] [Q&A] [answered]
Hello,

I'm using a FluentGrid like this:

```
<FluentDataGrid Items="@mUsers">
            <PropertyColumn Property="@(p => p.UserName)" Sortable="true" />
            <PropertyColumn Property="@(p => p.Email)" Sortable="true" />
        </FluentDataGrid>
```

This is working fine. But how can i display column content based on a contidion?

I've tried that. But it is not working:

```
<FluentDataGrid Items="@mUsers">
            @{
                if(context.IsLdapLogin==1) {
                    <TemplateColumn Title="Type">
                        <FluentIcon Icon="Icons.Regular.Size16.Remote" Color="@Color.Accent"/>
                    </TemplateColumn>
                } else {
                    <TemplateColumn Title="Type">
                        <FluentIcon Icon="Icons.Regular.Size16.Database" Color="@Color.Accent"/>
                    </TemplateColumn>
                }
            }
            <PropertyColumn Property="@(p => p.UserName)" Sortable="true" />
            <PropertyColumn Property="@(p => p.Email)" Sortable="true" />
        </FluentDataGrid>
```

I hope someone can help me...

Best Regards
Sascha

**Comment by vnbaaij** [comment]
There is an example for this in the demo site: https://www.fluentui-blazor.net/DataGrid#dynamiccolumns.

Razor code of that look suspiciously lot like yours though...I guess it is not working because of making it a code blok. Try it with the `@` directly on the `if` statement.

You could also do with one column and use the `FluentIcon` with the `Value` parameter to set it to an **instance** of an icon:
```
<FluentIcon Value="@stateicon" Color="@Color.Accent"/>
```
Where state icon is a variable sot to either `new Icons.Regular.Size16.Remote()`  or `new Icons.Regular.Size16.Database()`

**Reply by madcoda9000** [reply]
Hello,

thanks for your reply.

Changing the @ before the if statement is throwing the same error:

```
@if(context.IsLdapLogin) {
                <TemplateColumn>
                    <span>true</span>
                </TemplateColumn>
            } else {
                <TemplateColumn>
                    <span>true</span>
                </TemplateColumn>
            }
```

Message:
```
C:\Users\sheimann\source\repos\GroupVault\GroupVaultClient\Pages\Users.razor(28,17): error CS0103: The name 'context' does not exi 
st in the current context [C:\Users\sheimann\source\repos\GroupVault\GroupVaultClient\GroupVaultClient.csproj]
```

To me it seems, that context is not vailable.

Regarding your second hint: how should i set the correct state basesd on the value renedered currently by the fluentgrid?
Sure, i can create two variables holding different states and assign them to the column. But then i am not able to differ between the the states baes on the current value too.

**Reply by vnbaaij** [reply, marked as answer]
Ah, I didn't notice the use of `context`...What if you move the condition inside the `<TemplateColumn>`, i.e.:
```
<TemplateColumn>
    @if(context.IsLdapLogin) {
             <span>true</span>
           } else {
              <span>false</span>
     }
</TemplateColumn>
```
I' think `context` will then be of type `TGridItem` but not sure. Maybe you need to navigate up in your class /data model (ie something like `context.Row...`)


For second part. if you need to know data from the actual item that is not going to work I think. Like I sad, I did not notice the 'context' use

**Reply by madcoda9000** [reply]
Ahhhh, that's it! This way it is working.

This is how its displaying now:

![image](https://github.com/microsoft/fluentui-blazor/assets/25504739/6e5492fe-067b-419b-97d9-ff19ff1c51b9)



Best Regards
Sascha
